src/components/viewarticle_components/CommentCard.jsx

```diff
@@ -16,6 +21,19 @@ const CommentCard = ({ comment }) => {
     })
   }, [])
 
+  const deleteOnClickHandler = () => {
+    setProcessingDeleteRequest(true);
+    deleteCommentById(comment.comment_id).then((response) => {
+      if (response.name === "AxiosError") {
+        setFeedback(`Error: ${response.message}`);
+        setProcessingDeleteRequest(false);
+      } else {
+        setTriggerFetch((prevTriggerFetch) => !prevTriggerFetch);
```

Comment:
This does work but it doesn't do it optimistically because you are causing a refetch of the comments from the db and only once the comment has been deleted. 

To make this optimistic, you'll have to update the comments state before you perform the delete, giving the user an instant update. If the state is updated, the component will re-render. The user will see the 'local' update of state and then if they refresh, if the delete is successful, the comment will not show up

Reply:
Thanks for the detailed feedback. I'm on it👷‍♂️